### DISCUSSION TITLE
Merge blank screen and "Open channel" dialog

### DISCUSSION THREAD

**Discussion by x9136** [Chatterino/chatterino2#5115] [Ideas]
From a logical standpoint, there is nothing stopping you from combining these screens into one. And since this is one of the key functions, it will be nice to have 1 extra step removed.

![Screenshot 2024-01-21 at 08 24 41](https://github.com/Chatterino/chatterino2/assets/60105060/24f4b5a0-7e8b-4e75-80c7-023ad92eebcc)

![Screenshot 2024-01-21 at 08 24 49](https://github.com/Chatterino/chatterino2/assets/60105060/4dff1768-e6e6-4fc1-9961-54d9c203b678)

**Comment by pajlada** [comment]
We don't limit the size of a user's split, meaning this would be impractical.

**Reply by x9136** [reply]
I didn't quite get it. Is it impractical to write a hint in the dialog, for which you now need to take an extra step? If you look at it that way, it's impractical to keep the hint out of the dialog.

**Reply by pajlada** [reply]
The "click to open a dialog so you can select which channel should be in this split" text works. It requires an extra step - you're right in that, but because a split can be any size, it would be impractical to merge the dialog into the split. Not everything would fit, and would instead cause users to have to resize their window or their split to be able to join a channel.

**Reply by x9136** [reply]
> it would be impractical to merge the dialog into the split.

I agree with you, but I didn't suggest such a thing. My option was to write the hint inside the dialog.

**Reply by pajlada** [reply]
Which hint are you referring to?

**Reply by x9136** [reply]
This is what creating a new tab looks like to me:
https://kappa.lol/JylKt

Instead, clicking on the + could open a dialog like this:
![Screenshot 2024-04-17 at 10 25 38](https://github.com/Chatterino/chatterino2/assets/60105060/446c25d1-efff-4845-b095-3f3dc573f71c)

**Reply by pajlada** [reply]
Right, I misunderstood the entire point of the discussion then. "Merge the blank screen and the "Open Channel" dialog" is what confused me.

There's no real reason the hint is in one place and not the other, but I feel like there's probably a better place for it than the already crowded open channel dialog (e.g. some in-channel hint once you've opened a channel, since that will actually be on screen while you can use it).

**Reply by iProdigy** [reply]
initially also misunderstood this, but i can see a ux consistency argument where `+` to add a split to an existing tab *does* automatically show the dialog while `+` to create a new tab doesn't

somehow this never bothered me, but maybe I got accustomed to it
